posthog/api/insight.py

```diff
@@ -482,10 +482,11 @@ def insight_result(self, insight: Insight) -> InsightResult:
         dashboard_tile = self.dashboard_tile_from_context(insight, dashboard)
         target = insight if dashboard is None else dashboard_tile
 
-        refresh_insight_now, refresh_frequency = should_refresh_insight(insight, dashboard_tile)
-        if insight.filters and refresh_requested_by_client(self.context["request"]):
-            if refresh_insight_now:
-                return synchronously_update_cache(insight, dashboard, refresh_frequency)
+        refresh_insight_now, refresh_frequency = should_refresh_insight(
+            insight, dashboard_tile, request=self.context["request"], is_shared=self.context.get("is_shared")
+        )
+        if insight.filters and refresh_insight_now:
+            return synchronously_update_cache(insight, dashboard, refresh_frequency)
```

Comment:
flyby: I don't think we need `if insight.filters` here. 

`synchronously_update_cache` can handle insights that have queries instead of filters... In that case filters would be the empty object which I think is falsey in Python so we wouldn't refresh even though we could